[package] use digraph to back dependency visualization

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #57341 tmp intern
* #57340 [package] remove save_source_file API
* #57339 [package] remove PackageExporter.file_structure
* **#57338 [package] use digraph to back dependency visualization**
* #57337 [package] add simple graph data structure
* #57336 [package] make extern a dict
* #57335 [package] factor out `PackageExporter._get_dependencies`



Differential Revision: [D28114190](https://our.internmc.facebook.com/intern/diff/D28114190/)


Differential Revision: [D28114190](https://our.internmc.facebook.com/intern/diff/D28114190)

suo committed:
[package] use digraph to back dependency visualization

[ghstack-poisoned]

suo committed:
Update on "[package] use digraph to back dependency visualization"




Differential Revision: [D28114190](https://our.internmc.facebook.com/intern/diff/D28114190)

[ghstack-poisoned]

SplitInfinity commented on torch/package/_digraph.py:
```diff
@@ -54,6 +54,12 @@ def edges(self):
             for succ in successors:
                 yield n, succ
 
+    @property
+    def edges(self):
+        for n, successors in self._succ.items():
+            for succ in successors:
+                yield n, succ
+
```
```suggestion
```

SplitInfinity commented on torch/package/package_exporter.py:
```diff
@@ -313,7 +315,7 @@ def _module_exists(self, module_name: str) -> bool:
 
     def _write_dep_graph(self, failing_module=None):
-        edges = "\n".join(f'"{f}" -> "{t}";' for f, t in self.debug_deps)
+        edges = "\n".join(f'"{f}" -> "{t}";' for f, t in self.dependency_graph.edges)
```
```suggestion
        edges = "\n".join(f'"{from}" -> "{to}";' for from, to in self.dependency_graph.edges)
```

suo committed:
Update on "[package] use digraph to back dependency visualization"




Differential Revision: [D28114190](https://our.internmc.facebook.com/intern/diff/D28114190)

[ghstack-poisoned]

suo committed:
Update on "[package] use digraph to back dependency visualization"




Differential Revision: [D28114190](https://our.internmc.facebook.com/intern/diff/D28114190/)


Differential Revision: [D28114190](https://our.internmc.facebook.com/intern/diff/D28114190)

[ghstack-poisoned]

suo committed:
Update on "[package] use digraph to back dependency visualization"




Differential Revision: [D28114190](https://our.internmc.facebook.com/intern/diff/D28114190/)


Differential Revision: [D28114190](https://our.internmc.facebook.com/intern/diff/D28114190)

[ghstack-poisoned]

suo committed:
Update on "[package] use digraph to back dependency visualization"




Differential Revision: [D28114190](https://our.internmc.facebook.com/intern/diff/D28114190/)


Differential Revision: [D28114190](https://our.internmc.facebook.com/intern/diff/D28114190)

[ghstack-poisoned]

suo commented:
@suo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28114190).
